### DISCUSSION TITLE
How can I roll back to a previous version?

### DISCUSSION THREAD

**Discussion by dilectiogames** [Mikubill/sd-webui-controlnet#1803] [Q&A] [answered]
Anyone knows?

**Comment by huchenlei** [comment, marked as answer]
To checkout a previous commit in Git, you first need to find the commit hash. You can use the `git log` command to view the commit history. Once you have the commit hash, you can use the `git checkout` command to move to that commit. Here's how you can do it:

1. **Find the commit hash.**

   Use the following command to see the commit history:
   
   ```bash
   git log
   ```
   
   This command will show you a list of all commits. Each commit begins with a line that looks something like this:

   ```bash
   commit 1a2b3c4d5e6f...
   ```

   That alphanumeric string (e.g., `1a2b3c4d5e6f`) is the commit hash. Copy the commit hash of the commit you want to checkout.

2. **Checkout the commit.**

   Use the following command to checkout the commit:
   
   ```bash
   git checkout <commit-hash>
   ```

   Replace `<commit-hash>` with the actual commit hash that you copied.

Note: Checking out a commit will put your working directory in a "detached HEAD" state, which means you're not working on any branch. If you want to make changes and save them, you should create a new branch while you have that commit checked out:

```bash
git checkout -b <new-branch-name>
```

This will create a new branch with the name `<new-branch-name>` and check it out immediately.

Remember, you can always go back to your original branch using the `git checkout` command followed by the branch name:

```bash
git checkout <original-branch-name>
```

**Reply by dilectiogames** [reply]
Awesome, thank you.

I'm seeing only three commits with the last one from Mon Jul 17 22:06:14 2023 -0700.

any method to get version at least one or two weeks back?

if I git checkout to a previous version, how to get back to the last version?


I think I need to check this version

https://github.com/Mikubill/sd-webui-controlnet/commit/07bed6ccf8a468a45b2833cfdadc749927cbd575

is this: 07bed6ccf8a468a45b2833cfdadc749927cbd575 a commit number that I can use?

lol, yes it apears is the one ! :D will try now. see if I break everything!

**Reply by dilectiogames** [reply]
well going back was easy. I can't get to the current version :( arrhh!!

I get 

`*** Failed reading extension data from Git repository (sd-webui-controlnet)
    Traceback (most recent call last):
      File "E:\AI\stable-diffusion-webui\modules\extensions.py", line 65, in do_read_info_from_repo
        if repo.active_branch:
      File "E:\AI\stable-diffusion-webui\venv\lib\site-packages\git\repo\base.py", line 881, in active_branch
        return self.head.reference
      File "E:\AI\stable-diffusion-webui\venv\lib\site-packages\git\refs\symbolic.py", line 309, in _get_reference
        raise TypeError("%s is a detached symbolic reference as it points to %r" % (self, sha))
    TypeError: HEAD is a detached symbolic reference as it points to 'b4cdd1028346cc5e453c01e6b8336843a285c9fd'`

**Reply by huchenlei** [reply]
Note: Checking out a commit will put your working directory in a "detached HEAD" state, which means you're not working on any branch. If you want to make changes and save them, you should create a new branch while you have that commit checked out:
```
git checkout -b <new-branch-name>
```
This will create a new branch with the name <new-branch-name> and check it out immediately.

Remember, you can always go back to your original branch using the git checkout command followed by the branch name:
```
git checkout <original-branch-name>
```

In your case, original-branch-name should be main.

**Reply by dilectiogames** [reply]
worked with
git checkout main
Thank You!

**Reply by BooHooClaping** [reply]
hello why i cant git checkout commit hash?

**Reply by fadedfragger** [reply]
Hey. I wanna rollback to controlnet 1.1.440, but with "git log", i only see 1 hash.

edit: i think i found the hash (im not sure if it is eb451a007f7040288e865f96e9ee0842aa6ef91c), but how do i use it to get controlnet 1.1.440?

**Comment by TXMoon** [comment]
Where do I put this information?  You say "use the following command Get Log." but when I type that in the CMD prompt I get :
C:\Users\Name> get log
'get' is not recognized as an internal or external command, operable program or batch file.
D:\>git log
fatal: not a git repository (or any of the parent directories): .git
Why does all the responses, and searches on how to restore a previous version require me to be a MIT graduate programmer? Hell I am just a 1st grader with crayons on craft paper.
